torch/csrc/api/include/torch/types.h

```diff
@@ -7,6 +7,11 @@
 #include <torch/csrc/autograd/generated/variable_factories.h>
 #include <torch/csrc/autograd/variable.h>
 
+// TODO: These don't really belong here but torchvision builds in CI need them
+// Remove once the torchvision version being compiled in CI is updated
+#include <torch/library.h>
```

Comment:
This include causes problems for other builds. In particular, it prevents including torch.h direclty from what I can see:

```
In file included from original_header.h:7:
In file included from torch/csrc/api/include/torch/torch.h:3:
In file included from torch/csrc/api/include/torch/all.h:8:
In file included from torch/csrc/api/include/torch/data.h:3:
In file included from torch/csrc/api/include/torch/data/dataloader.h:3:
In file included from torch/csrc/api/include/torch/data/dataloader/stateful.h:3:
In file included from torch/csrc/api/include/torch/data/dataloader/base.h:3:
In file included from torch/csrc/api/include/torch/data/dataloader_options.h:4:
In file included from torch/csrc/api/include/torch/types.h:12:
xplat/caffe2/torch/library.h:397:39: error: no member named 'parseSchema' in namespace 'torch::jit'
  c10::FunctionSchema s = torch::jit::parseSchema(str);
                          ~~~~~~~~~~~~^
# More errors like that
```

Removing this include does allow the build to go smoothly.

Reply:
Looks like the include of `function_schema_parser.h` was guarded but the uses of it weren't. The old code probably included it transitively via `variable.h` or something like that.

Reply:
Thanks for finding this out!